### PR TITLE
lestarch: adding macOS to CI

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Workflow to run Framework, Ref, and Integration tests on macOS  using macos-latest runner.

### CODE REVIEW

File: .github/workflows/build-test-macos.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  MacOS-Integration:
+    runs-on: macos-latest
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+    - name: Install fprime tools
+      run: pip3 install fprime-tools fprime-gds
+    - name: Setup Dependencies
+      run: sudo apt-get install valgrind
```
Comment (Joshua-Anderson) [inline .github/workflows/build-test-macos.yml:59]: This isn't gonna work :)

Comment (LeStarch) [reply .github/workflows/build-test-macos.yml]: Yeah, I blew up on that one.  Should be fixed.

**Comment by LeStarch** [comment]
Woooooooo!

**Review by Joshua-Anderson — APPROVED** [review]
🚀

**Pull request merged**